arduino-ide-extension/src/node/utils/simple-buffer.ts

```diff
@@ -33,11 +35,39 @@ export class SimpleBuffer implements Disposable {
     Chunks.clear(this.chunks);
   }
 
+  private setTimeoutVariable(flushTimeout: number): void {
+    const isDisposed = this.disposed;
+    if (isDisposed) {
+      // once "isDisposed" is true we stop
+      // creating timeouts and do one more
+      // flush AFTER any setTimeout
+      // callback that may be in progress
+      this.flush();
+      return;
+    }
+
+    if (!this.flushTimeout) {
+      const onTimeout = () => {
+        this.flush();
+        this.clearTimeoutVariable();
+      };
+
+      this.flushTimeout = setTimeout(() => {
+        onTimeout();
+        this.setTimeoutVariable(flushTimeout);
+      }, flushTimeout);
+    }
+  }
+
+  private clearTimeoutVariable(): void {
+    if (this.flushTimeout) {
+      clearTimeout(this.flushTimeout);
+      this.flushTimeout = undefined;
+    }
+  }
+
   dispose(): void {
-    this.flush();
-    clearInterval(this.flushInterval);
-    this.clearChunks();
-    this.flushInterval = undefined;
+    this.disposed = true;
```

Comment:
alternatively we could use promises, rather than a flag, awaiting here and then performing a final flush

Reply:
Thanks for bringing it up. The flag is looking good. No need to over-engineer it.